zebra-network/src/protocol/external/codec.rs

```diff
@@ -184,10 +184,6 @@ impl Codec {
     /// Obtain the size of the body of a given message. This will match the
     /// number of bytes written to the writer provided to `write_body` for the
     /// same message.
-    ///
-    /// TODO: Replace with a size estimate, to avoid multiple serializations
-    /// for large data structures like lists, blocks, and transactions.
-    /// See #1774.
     fn body_length(&self, msg: &Message) -> usize {
```

Comment:
This is still a potential problem, but we don't want to fix it unless it's clear it's needed:

```suggestion
    // # Performance TODO
    //
    // If this code shows up in profiles, replace with a size estimate or cached size,
    // to avoid multiple serializations for large data structures like lists, blocks, and transactions.
    fn body_length(&self, msg: &Message) -> usize {
```